### PR TITLE
tokenbf_v1: fix token extraction from like expressions

### PR DESCRIPTION
**Pull request by filimonov**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

For changelog. Remove if this is non-significant change.

Category (leave one):
- Bug Fix

Short description (up to few sentences):
Trivial `LIKE '%string%'` expressions was not able to use tokenbf_v1 skip index.

Detailed description (optional):
In the case '%string%' token was cleared after coming to second percent sign.

BTW: just because % and _ are also ASCII and non-alphanumeric whole that if branch can be skipped

/cc @nikvas0

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Description for the changelog?

**Comment by alexey-milovidov** [comment]
A test?

**Comment by alexey-milovidov** [comment]
Why we pass `pos` by pointer not by reference?
According to our code style, we should always prefer references.

File: dbms/tests/queries/0_stateless/00945_tokenbf_v1_like_tokenizer_bug.sql [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    INDEX tok_bf (s, lower(s)) TYPE tokenbf_v1(512, 3, 0) GRANULARITY 1
+) ENGINE = MergeTree() ORDER BY id SETTINGS index_granularity = 8;
+
+insert into bloom_filter_like_tokenizer_bug select number, 'yyy,uuu' from numbers(1024);
+insert into bloom_filter_like_tokenizer_bug select number+2000, 'abc,def,zzz' from numbers(8);
+insert into bloom_filter_like_tokenizer_bug select number+3000, 'yyy,uuu' from numbers(1024);
+
+set max_rows_to_read = 16;
+
+SELECT max(id) FROM bloom_filter_like_tokenizer_bug WHERE s like '%,def,%';
```
Comment (filimonov) [inline dbms/tests/queries/0_stateless/00945_tokenbf_v1_like_tokenizer_bug.sql]: that was working

File: dbms/tests/queries/0_stateless/00945_tokenbf_v1_like_tokenizer_bug.sql [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+SELECT max(id) FROM bloom_filter_like_tokenizer_bug WHERE s like '%def%';
```
Comment (filimonov) [inline dbms/tests/queries/0_stateless/00945_tokenbf_v1_like_tokenizer_bug.sql]: and that - not

**Review by alexey-milovidov — APPROVED** [review]

**Comment by filimonov** [comment]
Hm. Looks like that fix didn't solve the problem. At least the test fails.

**Comment by filimonov** [comment]
It's by design behaviour - to make it possible token partial match.

**Comment by filimonov** [comment]
Closing it currently. Should be implemented other way.
Function like `hasToken` should be added.

**Comment by kiryaka** [comment]
It would be really nice to have hasToken analog for the array so that it would be possible to do a semi normal search like:
`select * from table where hasTokens(lower(payload), tokens('my search string'))`
something along the lines of `hasTokens` / `hasAnyTokens` / `hasAllTokens`

**Pull request closed without merging**